apps/previewgenerator.sh

```diff
@@ -16,67 +16,258 @@ debug_mode
+    done 11< results
+    rm -f results
+    clear      
+else
+    # check if imagick ist installed and remove it
+    if is_this_installed imagick
+    then
+        apt purge php-imagick -y
+    fi
+    # check if libmagickcore is installed and remove it
+    if is_this_installed libmagickcore-6.q16-3-extra
+    then
+        apt purge libmagickcore-6.q16-3-extra -y
+    fi    
+    # Choose file formats fo the case when imagick is not installed.
+    # for additional previews please look at the nextcloud documentation. But these probably won't work.
+    whiptail --title "Choose file formats" --checklist --separate-output "Now you can choose for which file formats you would like to generate previews\nSelect or unselect by pressing the spacebar" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    "PNG" "" ON \
+    "JPEG" "" ON \
+    "GIF" "" ON \
+    "BMP" "" ON \
+    "MarkDown" "" ON \
+    "MP3" "" ON \
+    "TXT" "" ON \
+    "Movie" "" OFF 2>results
+
+    while read -r -u 8 choice
+    do
+        case $choice in
+            "PNG")
+                occ_command config:system:set enabledPreviewProviders 11 --value="OC\\Preview\\PNG"
+            ;;
+
+            "JPEG")
+                occ_command config:system:set enabledPreviewProviders 12 --value="OC\\Preview\\JPEG"
+            ;;
+
+            "GIF")
+                occ_command config:system:set enabledPreviewProviders 13 --value="OC\\Preview\\GIF"
+            ;;
+
+            "BMP")
+                occ_command config:system:set enabledPreviewProviders 14 --value="OC\\Preview\\BMP"
+            ;;
+
+            "MarkDown")
+                occ_command config:system:set enabledPreviewProviders 15 --value="OC\\Preview\\MarkDown"
+            ;;
+
+            "MP3")
+                occ_command config:system:set enabledPreviewProviders 16 --value="OC\\Preview\\MP3"
+            ;;
+
+            "TXT")
+                occ_command config:system:set enabledPreviewProviders 17 --value="OC\\Preview\\TXT"
+            ;;
+
+            "Movie")
+                occ_command config:system:set enabledPreviewProviders 18 --value="OC\\Preview\\Movie"
+            ;;
+
+            *)
+            ;;
+        esac
+    done 8< results
+    rm -f results
+    clear
+fi
+
+# Set aspect ratio
+occ_command config:app:set previewgenerator squareSizes --value="32 256"
+occ_command config:app:set previewgenerator widthSizes  --value="256 384"
+occ_command config:app:set previewgenerator heightSizes --value="256"
+occ_command config:system:set preview_max_x --value="2048"
+occ_command config:system:set preview_max_y --value="2048"
+occ_command config:system:set jpeg_quality --value="60"
+occ_command config:app:set preview jpeg_quality --value="60"
+
+msg_box "In the last step you can define a specific Nextcloud user for which will be the user that runs the preview-generation. 
+
+The default behavoiur (just hit [ENTER]) is to run with the system user 'www-data' which will generate previews for all users. 
```

Comment:
hitting enter deosn't work here.
So I could remove that line?

Reply:
No please leave it. I'll test everything more later.

Reply:
Enter choose the default - `www-data`.